src/snowflake/cli/api/commands/snow_typer.py

```diff
@@ -111,3 +112,73 @@ def post_execute():
 
         log.debug("Executing command post execution callback")
         flush_telemetry()
+
+
+@dataclasses.dataclass
+class SnowTyperCommandData:
+    func: Callable
+    args: Tuple[Any, ...]
+    kwargs: Dict[str, Any]
+
+
+class SnowTyper:
+    """
+    SnowTyperApp factory. Usage is similar to typer.Typer, except that create_app()
+    creates actual SnowTyperInstance instance.
+    """
+
+    def __init__(
+        self,
+        /,
+        name: Optional[str] = None,
+        help: Optional[str] = None,  # noqa: A002
+        short_help: Optional[str] = None,
+        is_hidden: Optional[Callable[[], bool]] = None,
```

Comment:
```suggestion
        is_enabled: Optional[Callable[[], bool]] = None,
```
To be consistent with commands option

Reply:
discussed on Slack